src/app/password-input/password-input.component.ts

```diff
@@ -0,0 +1,14 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-password-input',
+  templateUrl: './password-input.component.html',
+  styleUrl: './password-input.component.css',
+})
+export class PasswordInputComponent {
+  showPassword: boolean = false;
+  togglePasswordVisibility() {
+    this.showPassword = !this.showPassword;
+  }
+  @Input() placeholder_text = '';
```

Comment:
it is better to use the same style code for variables (camelCase = placeholderText) 
properties together at the beginning of the class, then methods(please move placeholderText to showPassword)